src/core/server/http/http_service.ts

```diff
@@ -178,12 +195,9 @@ export class HttpService
 
   private async runNotReadyServer(config: HttpConfig) {
     this.log.debug('starting NotReady server');
-    const httpServer = new HttpServer(this.logger, 'NotReady');
-    const { server } = await httpServer.setup(config);
-    this.notReadyServer = server;
-    // use hapi server while KibanaResponseFactory doesn't allow specifying custom headers
-    // https://github.com/elastic/kibana/issues/33779
-    this.notReadyServer.route({
+    this.notReadyServer = new HttpServer(this.logger, 'NotReady');
+    const notReadySetup = await this.notReadyServer.setup(config);
+    notReadySetup.server.route({
```

Comment:
Let's use newly added API instead of hapi one?

Reply:
I was going to do that, but since we don't support some of the HTTP methods right now, I left it like this so that it covers _any_ request.